sonar-helm-for-iac/converters/data-loading.go

```diff
@@ -0,0 +1,99 @@
+package converters
+
+import (
+	"fmt"
+	"github.com/samber/mo"
+	"sigs.k8s.io/yaml"
+)
+
+// TemplateSources contains all the sources needed to evaluate a template
+type TemplateSources struct {
+	Name                  string
+	RawTemplate           string
+	fileNameToFileContent map[string]string
+}
+
+func NewTemplateSources(name string, rawTemplate string, fileNameToFileContent map[string]string) *TemplateSources {
+	return &TemplateSources{name, rawTemplate, fileNameToFileContent}
+}
+
+// NumAdditionalSources returns the number of sources required for this template.
+func (ts *TemplateSources) NumAdditionalSources() int {
+	return len(ts.fileNameToFileContent)
+}
+
+func (ts *TemplateSources) SourceFile(name string) (string, error) {
+	if _, ok := ts.fileNameToFileContent[name]; !ok {
+		return "", fmt.Errorf("source file %s not found", name)
+	}
+	return ts.fileNameToFileContent[name], nil
+}
+
+func (ts *TemplateSources) Values() string {
+	valuesFile, _ := ts.SourceFile("values.yaml")
+	return valuesFile
+}
+
+func PrepareChartValues(templateSources *TemplateSources) (map[string]interface{}, error) {
+	result := map[string]interface{}{}
+
+	values := mo.TupleToResult(LoadValues(templateSources.Values()))
+	chart, err := Map(values, func(values *Values) (*Chart, error) {
+		return Map(mo.TupleToResult(templateSources.SourceFile("Chart.yaml")), func(content string) (*Chart, error) {
+			return LoadChart(content)
+		}).Get()
+	}).Get()
```

Comment:
Why not evaluate this function prior to the return?

Reply:
For now this approach with `mo.Result` seems to me a bit better than vanilla error handling in Go, which in this case would require multiple repetitive blocks, because several steps here can return errors preventing the evaluation. So wrapping everything into a chain of `Result`s is a way to handle  errors in a single place.